modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/communicators/aah_async_comm.h

```diff
@@ -17,46 +17,51 @@ namespace lbs
 class FLUDS;
 
 /**
- * Handles the swift communication of interprocess communication related to sweeping.
+ * Handles interprocess communication related to sweeping.
  */
 class AAH_ASynchronousCommunicator : public AsynchronousCommunicator
 {
 private:
-  int num_groups_;
-  int num_angles_;
-  int max_num_messages_;
-  int max_mpi_message_size_;
+  size_t num_groups_;
+  size_t num_angles_;
+  size_t max_num_messages_;
+  size_t max_mpi_message_size_;
```

Comment:
I seem to recall a discussion about a distaste for `size_t` and a preference towards `int`. Any particular reason for this?

Reply:
Yes, that is still an issue that we need to resolve. Probably sooner rather than later. I always like to represent sizes of things like  buffers, memory, messages, etc. as size_t, so that was why I promoted max_num_messages and max_mpi_messages. Since we don't have a clear decision on group and angles, I promoted them to prevent potential casts that might need to occur in computing the number of messages and the message size. When we have a firm decision on how we want to handle those quantities, I can fix this as necessary.